File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        int ind=-1;
        
        for(int i=0;i<word.size();i++){
            if(ch==word[i]){
                ind=i;
                break;
            }
        }
        
        if(ind!=-1)reverse(word.begin(),word.begin()+ind+1);
       
        return word;
    }
};

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    
    public static String swap(String s, int x, int y){
        StringBuilder sb = new StringBuilder(s);
        sb.setCharAt(x, s.charAt(y));
        sb.setCharAt(y, s.charAt(x));
        return sb.toString();
    }
    
    public String reversePrefix(String word, char ch) {
        int ind=-1;
        
        for(int i=0;i<word.length();i++){
            if(ch==word.charAt(i)){
                ind=i;
                break;
            }
        }
        
        int s=0,e=ind;
        
        while(s<e){
            word = swap(word,s,e);
            s++;e--;
        }
        
        return word;
    }
}